app/containers/mainPanel.js

```diff
@@ -0,0 +1,137 @@
+import React, { Fragment, Component } from 'react'
+import { clipboard, ipcRenderer } from 'electron'
+import { connect } from 'react-redux'
+
+import { changeScreen, getMessages, showCabalSettings, viewCabal } from '../actions'
+import CabalSettings from './cabalSettings'
+import WriteContainer from './write'
+import MessagesContainer from './messages'
+
+const mapStateToProps = state => ({
+  addr: state.currentCabal,
+  cabal: state.cabals[state.currentCabal],
+  cabals: state.cabals,
+  cabalSettingsVisible: state.cabalSettingsVisible
+})
+
+const mapDispatchToProps = dispatch => ({
+  getMessages: ({ addr, channel, count }) => dispatch(getMessages({ addr, channel, count })),
+  changeScreen: ({ screen, addr }) => dispatch(changeScreen({ screen, addr })),
+  viewCabal: ({ addr }) => dispatch(viewCabal({ addr })),
+  showCabalSettings: ({ key }) => dispatch(showCabalSettings({ key })),
+})
+
+class MainPanel extends Component {
+  constructor (props) {
+    super(props)
+    this.state = { showEmojiPicker: false }
+    this.shouldAutoScroll = true
+    this.scrollTop = 0
+    this.composerHeight = 55
+    this.toggleEmoji = this.toggleEmoji.bind(this)
+    this.handleOpenCabalUrl = this.handleOpenCabalUrl.bind(this)
+  }
+
+  componentDidMount () {
+    var messagesDiv = document.querySelector('.messages')
+    if (messagesDiv) messagesDiv.scrollTop = this.scrollTop
+    ipcRenderer.on('open-cabal-url', (event, arg) => {
+      this.handleOpenCabalUrl(arg)
+    })
+  }
+
+  componentDidUpdate (prevProps) {
+    if (prevProps.cabal != this.props.cabal) {
+      this.scrollToBottom()
+    }
+    // if you're in the same cabal and a new message arrives we should show a button prompting a scroll to bottom
+  }
+
+  handleOpenCabalUrl ({ url = '' }) {
+    let addr = url.replace('cabal://', '').trim()
+    if (this.props.cabals[addr]) {
+      this.props.viewCabal({ addr })
```

Comment:
You changed 'addr' to key in the backend, but it can be helpful in dynamic languages like node to use consistent naming across the codebase when referring to the same type of object.

Reply:
(I realize this is out of the scope of this particular PR, but something to keep in mind. Perhaps we could open an issue?)

Reply:
Great point! I've reverted where I've used key in this pr to use addr. Though I'd like to change all of the addr to key at some point to be consistent with the cli.